lib/src/component_declaration/builder_helpers.dart

```diff
@@ -134,6 +134,84 @@ abstract class UiProps extends component_base.UiProps with GeneratedClass {
   @toBeGenerated PropsMetaCollection get staticMeta => throw UngeneratedError(member: #meta);
 }
 
+/// Helper static extension methods to make forwarding props easier.
+extension PropsToForward<T extends UiProps> on T {
+
+  /// Returns a copy of this instance's props excluding the keys found in [exclude].
+  ///
+  /// [exclude] should be a `Set` of PropsMixin `Type`s.
+  /// If [exclude] is not set it defaults to using the current instances Type.
+  ///
+  /// __Example:__
+  ///
+  /// ```dart
+  /// // within a functional component: `uiFunction<FooPropsMixin>`
```

Comment:
I think it'd be helpful, at the expense of verbosity, to show the full component to help clarify what's going on:

> Component with a single props mixin:
> ```dart
> mixin FooPropsMixin on UiProps {
>   String foo;
> }
> 
> UiFactory<FooPropsMixin> Foo = uiFunction((props) {
>   return (Bar()
>     // Filter out props declared in FooPropsMixin 
>     // (used as the default for `exclude` since that's what `props` is statically typed as) 
>     // when forwarding to Bar.
>     ..addAll(props.getPropsToForward())
>   )();
> });
> ``` 
>
> Component with a more than one props mixin:
> ```dart
> mixin FooPropsMixin on UiProps {
>   String foo;
> }
> class FooProps = UiProps with BarProps, FooPropsMixin;
> 
> UiFactory<FooPropsMixin> Foo = uiFunction((props) {
>   return (Bar()
>     // Filter out props declared in FooPropsMixin when forwarding to Bar.
>     ..addAll(props.getPropsToForward({FooPropsMixin}))
>   )();
> });
> ```